project/frontend/src/pages/HomePage.js

```diff
@@ -1,21 +1,27 @@
 import './HomePage.css';
 
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, { useState } from 'react';
+
+import Modal from '../components/Modal.js';
+import PreferenceForm from '../components/PreferenceForm.js';
 
 function HomePage() {
-  return (
-    <div className='container u-full-width'>
+  const [isPreferenceFormOpen, setIsPreferenceFormOpen] = useState(false);
+
+  const togglePreferenceForm = () => {
+    setIsPreferenceFormOpen((prev) => !prev);
```

Comment:
how come you can't you just do setIsPreferenceFormOpen(!prev)?

Reply:
setState functions (from a `useState` hook or `setState` in a class component) are asynchronous. So if a `setState` call relies on the previous state it is better to pass a function to `setState` than to pass an object.

This page from Reactjs.org explains it better: https://reactjs.org/docs/faq-state.html. That page is about class components but from what I've read the same thing applies to using the `useState` hook like I've done in this component.